### PR TITLE
CI: Arch build via Docker

### PR DESCRIPTION
**Pull request by bryphe**

### CODE REVIEW

**Comment by tcoopman** [comment]
I might find some time to look at this tomorrow if you can give me pointers on what you want to test / improve.

**Comment by bryphe** [comment]
@tcoopman - that would be awesome! Thank you for the help!

There were a few things I was trying to look at:

First, with this build, I wanted to see if I could reproduce the failure on Arch Linux from the Onivim 2 issue: https://github.com/onivim/oni2/issues/334 - that would be a good indication that the Docker environment is a fair representation of an Arch development environment. Unfortunately - that doesn't seem to be the case - it's actually building! In other words, that `acl_get_file`/`acl_set_file`/`acl_free` issue isn't reproducing in this Docker environment.

Second, of course, I want to fix the actual build issue 😄 The codepath that is causing the problem is here: ```
#ifdef HAVE_POSIX_ACL
    ret = (vim_acl_T)acl_get_file((char *)fname, ACL_TYPE_ACCESS);
#else
``` (https://github.com/onivim/libvim/blob/359e46fe583d7c92ca2ffe58a598c5ab74411f26/src/os_unix.c#L2231)

It looks like, on Arch, we may need the `acl` library (`pacman -S acl`). Alternatively - it might be that there is a different ACL library we should be using out-of-the-box in Arch? In that case, we'll need to update the configure step to pick up the right one.

File: src/build/docker/arch/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+RUN pacman --noconfirm -S nodejs
+RUN pacman --noconfirm -S npm
+RUN pacman --noconfirm -S gcc
+RUN pacman --noconfirm -S coreutils
+RUN pacman --noconfirm -S grep
+RUN pacman --noconfirm -S tar
+RUN pacman --noconfirm -S sed
+RUN pacman --noconfirm -S gawk
+RUN pacman --noconfirm -S make
+RUN pacman --noconfirm -S diffutils
```
Comment (tcoopman) [inline src/build/docker/arch/Dockerfile]: Is there a reason that you write every dependency as a separate line vs `pacman --noconfirm -S nodejs npm gcc ...`?

Comment (bryphe) [reply src/build/docker/arch/Dockerfile]: @tcoopman - good catch! No reason to have separate lines - your suggestion is cleaner 👍

**Pull request merged**